### PR TITLE
feat(nuxt): native async-context in vue's `withAsyncContext`

### PR DESCRIPTION
**Pull request by pi0**
### 🔗 Linked issue

Followup on #20918 

[reproduction](https://stackblitz.com/edit/github-6izg8q?file=nuxt.config.ts) by @Atinux confirming issue (should run locally)

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR, tries an alternative method of integrating Native Async Context with Vue by directly patching `withAsyncContext` import that is injected by Vue for it's non native async context support.

This method has the benefit of being much simpler as we don't need AST based transforms also probably more future compliance because reflects the ideal change we expect from Vue upstream support to improve internal `withAsyncContext`.

~~Note: I could confirm this solution externally on reproduction but seems local fixture is not working, investigating more. -- Update: It strangely also works fine in `playground` but not fixture still!~~ (working both externally and in fixture 🚀 )

--- 

Local module to try: 

```ts
import { defineNuxtModule, addBuildPlugin } from 'nuxt/kit'
import { createUnplugin } from 'unplugin'

export default defineNuxtModule({
  setup(_, nuxt) {
    addBuildPlugin(createUnplugin(() => {
      const virtualFileId = '\0vue-async-context'
      return {
        name: 'nuxt:vue-async-context',
        transformInclude (id) {
          return id.endsWith('.vue')
        },
        resolveId (id) {
          if (id === virtualFileId) {
            return id
          }
        },
        load(id) {
          if (id !== virtualFileId) {
            return
          }
          return `
          import { withAsyncContext as withVueAsyncContext, getCurrentInstance } from 'vue'
          export function withAsyncContext(fn) {
            return withVueAsyncContext(() => {
              const nuxtApp = getCurrentInstance()?.appContext.app.$nuxt
              return nuxtApp ? nuxtApp.runWithContext(fn) : fn()
            })
          }
          `.trim()
        },
        transform (code) {
          if (!code.includes('_withAsyncContext')) {
            return
          }
          code = `import { withAsyncContext as _withAsyncContext } from "${virtualFileId}"
                  ${code.replace('withAsyncContext as _withAsyncContext', '')}`
          return {
            code,
            map: null
          }
        }
      }
    }))
  }
})
```

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
/cc @antfu Seems Webpack VFS with unplugin is not working well. I couldn't reproduce it externally (https://stackblitz.com/edit/stackblitz-starters-c3p5u7) so pinging you if you can spot any obvious error in usage.

(To reproduce on branch `TEST_BUILDER=webpack TEST_CONTEXT=async pnpm nuxt dev test/fixtures/basic`)

File: packages/nuxt/src/core/plugins/async-context.ts [hunk 1]
```diff
@@ -1,46 +1,21 @@
...
-                  }
-                }
-              }
-            })
-          }
-        }
-      })
-
+      s.prepend('import { withAsyncContext as _withAsyncContext } from "#app/composables/asyncContext";\n')
+      s.replace(/withAsyncContext as _withAsyncContext,?/, '')
```
Comment (danielroe) [inline packages/nuxt/src/core/plugins/async-context.ts:18]: should this be:

```suggestion
      s.replace(/withAsyncContext as _withAsyncContext,?/g, '')
```

**Comment by Atinux** [comment]
I can confirm it works locally for the custom module 💯

**Review by atinux — APPROVED** [review]
This is awesome.

**Comment by Atinux** [comment]
Shall we add a section on https://nuxt.com/docs/guide/concepts/auto-imports#using-vue-and-nuxt-composables to explain our experimental feature, otherwise I don't know how they can test this 😅

**Review by danielroe — APPROVED** [review]

**Comment by joffreyBerrier** [comment]
Hi @pi0 do you have an idea of that https://github.com/nuxt/nuxt/issues/23777 ?

**Pull request merged**